functions.ipynb

```diff
@@ -405,6 +415,107 @@
   {
    "cell_type": "markdown",
    "id": "25",
+   "metadata": {},
+   "source": [
+    "---"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "26",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%reload_ext tutorial.tests.testsuite"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "27",
+   "metadata": {},
+   "source": [
+    "## Exercise 1"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "28",
+   "metadata": {},
+   "source": [
+    "Write a Python function called `greet` that takes two parameters: `name` (a string) and `age` (an integer).\n",
+    "The function should return a greeting message in the following format: `\"Hello, <name>! You are <age> years old.\"`\n",
+    "\n",
+    "<div class=\"alert alert-block alert-warning\">\n",
+    "    <h4><b>Note</b></h4>\n",
+    "    <strong>Do not</strong> forget to write a proper docstring and add the correct type hints to the parameters and the return value.\n",
+    "</div>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "29",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%ipytest\n",
+    "\n",
+    "def solution_greet():\n",
+    "    return"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "30",
+   "metadata": {},
+   "source": [
+    "## Exercise 2\n",
+    "\n",
+    "Write a Python function called `calculate_area` that takes three parameters: `length` (a float), `width` (a float), and `unit` (a string with a **default** value of `\"cm\"`).\n",
+    "The function should calculate the area of a rectangle based on the given length and width, and return the result **as a tuple** with the correct, default unit (i.e., `cm^2`).\n",
+    "If the unit parameter is \"m\", the function should convert the length and width from meters to centimeters before calculating the area.\n",
+    "\n",
+    "Your solution function **must** handle the following input units (the output unit is **always** `cm^2`):\n",
+    "\n",
+    "- centimeters (`cm`)\n",
+    "- meters (`m`)\n",
+    "- millimeters (`mm`)\n",
+    "- yards (`yd`)\n",
+    "- feet (`ft`)\n",
+    "\n",
+    "If you pass an unsupported unit, the function should **return** a string with the error message: `Invalid unit: <unit>`, where `<unit>` is the unsupported unit.\n",
+    "\n",
+    "<div class=\"alert alert-block alert-warning\">\n",
+    "    <h4><b>Note</b></h4>\n",
+    "    <strong>Do not</strong> forget to write a proper docstring and add the correct type hints to the parameters and the return value.\n",
+    "</div>"
```

Comment:
```suggestion
    "</div>"
    <div class="alert alert-block alert-info">
    <h4><b>Hints</b></h4>
    <ul>
        <li>1 yd = 91.44 cm</li>
        <li>1 ft = 30.48 cm</li>
    </ul>
</div>
```

Reply:
Just be careful when editing HTML inside notebooks: forgetting or misplacing a tag breaks everything.

Reply:
You are right, I forgot the double quotes 😅